bin/test_pyodide.mjs

```diff
@@ -12,20 +12,65 @@ for (const fileName of fileNames) {
 
 const pyodide = await pyodide_pkg.loadPyodide()
 await pyodide.loadPackage([
-    'mpmath',  // provided by pyodide
-    'numpy',  // built by pyodide
-    `../dist/${sympy}`  // git version sympy
+    'micropip',
+    'mpmath',  // provided by Pyodide
+    'numpy',  // built by Pyodide
+    'pytest',  // provided by Pyodide
+    `./dist/${sympy}`  // git version SymPy
 ])
+console.log('Pyodide packages loaded successfully')
 
-let split = 'None'
+const micropip = pyodide.pyimport('micropip');
+await micropip.install('pytest-split');
+console.log('Micropip packages loaded successfully')
+
+let group = 'None'
 if (argv[2]) {
-    if (argv[2].startsWith('--split=')) {
-        split = `'${argv[2].slice(8)}'`
+    if (argv[2].startsWith('--group=')) {
+        group = argv[2].slice(8)
+    }
+}
+let splits = 'None'
+if (argv[3]) {
+    if (argv[3].startsWith('--splits=')) {
+        splits = argv[3].slice(9)
     }
 }
+let sympyInstallPath = '/lib/python3.10/site-packages/sympy'
+console.log('Variables set successfully')
 
 pyodide.runPython(`
-import sympy
-if not sympy.test(split=${split}, subprocess=False):
-    exit(1)
+import os
+import sys
+
+import pytest
+
+os.chdir('${sympyInstallPath}')
+
+args = [
+    '--rootdir',
+    '${sympyInstallPath}',
+    '-m',
+    'not slow',
+    '--group',
+    '${group}',
+    '--splits',
+    '${splits}',
+    '--durations',
+    '20',
+    '--ignore',
+    '${sympyInstallPath}/integrals/rubi/rubi_tests/tests',
+    '--ignore',
+    '${sympyInstallPath}/testing/tests/test_runtests_pytest.py',
+    '-W',
+    'ignore::pytest.PytestUnknownMarkWarning',
+    '${sympyInstallPath}',
+]
+exit_code = pytest.main(args)
+
+if exit_code != pytest.ExitCode.OK:
+    print(f'Exiting with pytest exit code {exit_code}')
+    sys.exit(exit_code.value)
+else:
+    print(f'pytest finished with exit code {exit_code}')
 `)
```

Comment:
Not that I want to reopen this can of worms but did you try adding a `sys.exit` call right underneath this line?

I think I remember that you did.

This line is printed out in the CI jobs that timeout but there's nothing else here so `pyodide` should exit.